### ISSUE TITLE
WebJobs not running

### ISSUE THREAD

**Issue by ypcode** [SharePoint/PnP-Partner-Pack#206]
Hello,

I followed the setup procedure a few weeks ago just for trying, everything worked fine. I then activated the Azure trial subscription for the domain of my developer tenant and it is now expired.
Do all the assets (App service, storage accounts) have to be installed in the same subscription as the Azure AD one ?

Actually, I tried it with a valid Azure subscription we have but in another domain and get the following messge in the web jobs log : 

Microsoft.Azure.WebJobs.Host.FunctionInvocationException: Microsoft.Azure.WebJobs.Host.FunctionInvocationException: Exception while executing function: Functions.ProcessQueueMessage ---> Microsoft.IdentityModel.Clients.ActiveDirectory.AdalServiceException: AADSTS90002: No service namespace named 'xxxdev.sharepoint.com' was found in the data store. Trace ID: 0aaccd61-1f59-4e21-9bb7-393883dcf49f Correlation ID: fff53cd6-c5dc-4f77-8388-0b9fb26ea140 Timestamp: 2017-01-13 17:26:17Z ---> System.Net.WebException: The remote server returned an error: (400) Bad Request. at System.Net.HttpWebRequest.GetResponse() at Microsoft.IdentityModel.Clients.ActiveDirectory.HttpWebRequestWrapper.<GetResponseSyncOrAsync>d__2.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Microsoft.IdentityModel.Clients.ActiveDirectory.HttpHelper.<SendPostRequestAndDeserializeJsonResponseAsync>d__0`1.MoveNext() --- End of inner exception stack trace --- at Microsoft.IdentityModel.Clients.ActiveDirectory.AuthenticationContext.RunAsyncTask[T](Task`1 task) at Microsoft.IdentityModel.Clients.ActiveDirectory.AuthenticationContext.AcquireToken(String resource, ClientAssertionCertificate clientCertificate) at OfficeDevPnP.Core.AuthenticationManager.<>c__DisplayClass26_0.<GetAzureADAppOnlyAuthenticatedContext>b__0(Object sender, WebRequestEventArgs args) at Microsoft.SharePoint.Client.ClientRuntimeContext.OnExecutingWebRequest(WebRequestEventArgs args) at Microsoft.SharePoint.Client.ClientContext.GetFormDigestInfoPrivate() at Microsoft.SharePoint.Client.ClientContext.EnsureFormDigest() at Microsoft.SharePoint.Client.ClientContext.ExecuteQuery() at Microsoft.SharePoint.Client.ClientContextExtensions.ExecuteQueryImplementation(ClientRuntimeContext clientContext, Int32 retryCount, Int32 delay) at Microsoft.SharePoint.Client.ClientContextExtensions.ExecuteQueryRetry(ClientRuntimeContext clientContext, Int32 retryCount, Int32 delay) at OfficeDevPnP.PartnerPack.Infrastructure.SharePoint.SharePointProvisioningRepository.GetProvisioningJob(Guid jobId, Boolean includeStream) in 

Seems to be an authentication error.
However i double checked the certificates, everything looks fine...

Could you please tell me if I DO NEED to install everything within the "same" Azure ?

EDIT: 
===

So, I finally tried to reinstall the whole thing on the subscription of my SharePoint tenant, and the problem is still EXACTLY the same, I don't know if it is related to a misconfiguration or something else.

From the UI, I can query and post the SharePoint infra site data, it would mean there is no authentication issue there...

However, I noticed something else as well that seem to have already been reported, I have 5 test users in  my developer tenant but I cannot find them from the people picker in the Partner Pack CreateSiteCollection page, I can only find my own account that is the tenant admin. Maybe it is unrelated, but maybe not

**Comment by PaoloPia** [comment]
Hi,
You can setup the azure resources in whatever tenant you like, as long as you do the application registration in the Azure AD tenant under the cover of the Office 365 tenant.
I don't know the details of your setup process, but it looks like you are targeting a wrong tenant. I would double-check the .config files and the application registration in Azure AD.

**Comment by ypcode** [comment]
Hi Paolo, 

I had been reviewing all the procedure for several hours and every single argument and always missed that.
the domain I obfuscated in the error message above is the correct name of my sharepoint tenant, so I was considering it as correct,
Since it was actually expecting the **AD Tenant**, which totally makes sense, the argument must be **xxxdev.onmicrosoft.com** and not **xxxdev.sharepoint.com**

However, I don't get how it could work in the web application with the wrong settings, for instance. to put the jobs in the list of my infra site collection, because the settings were the same in all the app and web.config files (done via your powershell script). And, in both cases it is actually querying the same SharePoint site collection (infra) since the error occured in OfficeDevPnP.PartnerPack.Infrastructure.SharePoint.SharePointProvisioningRepository.GetProvisioningJob

However, this issue is now solved.
Thanks for your help !
